### ISSUE TITLE
Tabs (admin menu tabs for ModuleAdminController)

### ISSUE THREAD

**Issue by mortenhartvig** [PrestaShop/PrestaShop#10164]
**Describe the bug**
When creating a ModuleAdminController, and you want it to appear in the admin menu (via the new 1.7.1.x support for $tabs array in the module main class), it is unclear on how to precicely create it, and there is also missing feedback on template errors (hard to backtrack, as it may be present in multiple locations. Documentation could solve).

Example given:

Say you have a **AdminSomeNameController.php** in modules/somename/controllers/admin, and you want to create a menu tab for it.

**Issue 1:** 
In the given tabs property (in the main class), the class_name is very unclear. The below works, but it is unclear if it should be called either 'SomeName', AdminSomeName (correct), or 'AdminSomeNameController'

`public $tabs = array(
        array(
            'name' => 'Random Name',
            'class_name' => 'AdminSomeName,
            'visible' => true,
            'parent_class_name' => 'SELL',
        )
    );`

**Issue 2:**

If you create a template, that is called from renderview() in the above AdminSomeNameController, it is very hard to figure out why it dosn't work.

What i'm saying is, that the following (correct) path for template.tpl, is very hard to guess for a first time module developer: 
modules/somename/views/templates/admin/**some_name**/template.tpl.

(the snake_case module name inside /views/templates/admin is not logical, and requires reading the actual native code to figure it out). 

Maybe describing the template/renderview() part here would be enough:
http://doc.prestashop.com/display/PS17/Adding+module+links+in+the+back-office+side+menu

**Additionnal information**
PrestaShop version: 1.7.2.2 (relevant for > 1.7.1.x)
PHP version: N/A

**Comment by mortenhartvig** [comment]
TLDR:  :-)

A: Document class_name in the $tabs array 
Unclear what is the correct name: 
'SomeName', AdminSomeName (correct), or 'AdminSomeNameController'

B: Guessing correct template path/structure is very hard (requires reading the native code)
'modules/somename/views/templates/admin/some_name/template.tpl.'

A and B could be solved in the doc:
http://doc.prestashop.com/display/PS17/Adding+module+links+in+the+back-office+side+menu

**Comment by khouloudbelguith** [comment]
Hi @mortenhartvig,

Would you be willing to make a pull request on GitHub with your suggestion?
https://github.com/PrestaShop/docs
Thank you!

Best regards, Khouloud

**Comment by Quetzacoalt91** [comment]
The first part can be fixed with https://github.com/PrestaShop/docs/pull/126

The second part needs to be completely written, we only describe the front controllers on the current dev docs (https://devdocs.prestashop.com/1.7/modules/concepts/controllers/front-controllers/).

**Comment by khouloudbelguith** [comment]
Hi @Quetzacoalt91,

Thanks for your fix.

Best regards, Khouloud
